### PR TITLE
Survival capsules no longer get deleted when deployed at the edge of the map

### LINKED ISSUE #55674: bluespace capsules used on the edges of lavaland do not expand, yet are consumed.
## Reproduction:

Throw a capsule at the edge of lavaland
observe that the capsule does not activate, yet disappears.

### PR DESCRIPTION
**Pull request by dexmoh**
## About The Pull Request
Fixes #55674

## Changelog
:cl: Dex
fix: Survival capsules no longer get deleted when deployed at the edge of the map.
/:cl:

### CODE REVIEW

File: code/modules/mining/equipment/survival_pod.dm [hunk 1]
```diff
@@ -53,20 +53,20 @@
...
-		playsound(src, 'sound/effects/phasein.ogg', 100, TRUE)
-
-		var/turf/T = deploy_location
-		if(!is_mining_level(T.z)) //only report capsules away from the mining/lavaland level
-			message_admins("[ADMIN_LOOKUPFLW(usr)] activated a bluespace capsule away from the mining level! [ADMIN_VERBOSEJMP(T)]")
-			log_admin("[key_name(usr)] activated a bluespace capsule away from the mining level at [AREACOORD(T)]")
-		template.load(deploy_location, centered = TRUE)
-		new /obj/effect/particle_effect/smoke(get_turf(src))
-		qdel(src)
+		if(template.load(deploy_location, centered = TRUE)) // Did capsule deploy?
```
Comment (coiax) [inline code/modules/mining/equipment/survival_pod.dm:59]: This check should be part of `/datum/map_template/proc/check_deploy()`.

**Review by coiax — CHANGES_REQUESTED** [review]

File: code/modules/mining/equipment/survival_pod.dm (hunk 1 above)
Comment (Ewwmewgewd) [reply code/modules/mining/equipment/survival_pod.dm]: Resolved.

**Comment by LemonInTheDark** [comment]
How does it work now exactly?

**Comment by Ewwmewgewd** [comment]
> How does it work now exactly?

@LemonInTheDark inside `proc/check_deploy()` I added this block of code that checks whether the capsule sticks out of the map. It takes the location (center), width and height of the capsule and it calculates the location of the lower-left point of the capsule. Based on that location it check whether the thing sticks out of the map.

Capsule will stick out of the map if `shelter_origin_x` is less than 1 or higher or equal to `world.maxx`. Apply the same logic to the y coordinate.

**Pull request merged**